### DISCUSSION TITLE
Update version of PySr

### DISCUSSION THREAD

**Discussion by AKHCE** [MilesCranmer/PySR#349] [Q&A] [answered]
Hi,
I want to update the PySr package. I went through all the installation steps from scratch. For example, I update Julia and then PySr. But it seems that the old version is still running.
I am wondering if there is any straightforward way to update PySr.
Thanks

**Comment by MilesCranmer** [comment]
More info needed: 
- Are you using pip or conda?
- What did you try to update it?
- How did you install Julia

**Reply by AKHCE** [reply]
Thank you so much for your reply. 
-- I tried pip.
-- The version installed on my machine is '0.11.3', I want to upgrade it to the latest version i.e. '0.14.3'
-- I download from the Julia's website and install it as admin and add it to the path as well.

**Reply by MilesCranmer** [reply]
How did you try to update PySR?

(if unclear, the method to use is `pip install -U pysr` and then `python -m pysr install`)

**Reply by AKHCE** [reply]
![1](https://github.com/MilesCranmer/PySR/assets/122554992/c9663c52-380d-43dd-b7e8-33d5116b90a9)
![2](https://github.com/MilesCranmer/PySR/assets/122554992/ee772cc4-e433-480c-8808-a02f5cbb1384)
Thanks. Yes, I did that exactly. 
As you can see in the attached, the wired thing happened is that 0.14.3 is already installed in the base environment, I checked by conda list, see pic 1. 
But once I import pysr and check the version, see pic 2, the version is still 0.11.3!!

It seems installed successfully but once imported, it still uses old version i.e. 0.11.3.

**Reply by MilesCranmer** [reply]
`conda uninstall pysr`?

**Reply by AKHCE** [reply, marked as answer]
Finally, I resolved the issue.
What I did was; first, I found the address where the old pysr lives by using "pysr._ _ path _ _" . Then, I manually deleted the old version and installed the new version. 

Thanks

**Reply by AKHCE** [reply]
Although I succeeded to update my pysr package, I now have issue in running it. please see the screenshot attached.  I did add julia to the path, also I manually copied julia's bin folder to the paths (os.environ(["PATH"])).

Any advice is so much appreciated.

**Reply by MilesCranmer** [reply]
Did you install Julia through conda? If so, you have to also install PySR through conda. i.e., there are two ways:

1. Manual julia install on your system (I like [juliaup](https://github.com/JuliaLang/juliaup)), then pip install of PySR.
2. Conda to install everything.

Any mix of these is going to cause an issue, so just do 1 or 2.

**Reply by AKHCE** [reply]
Thank you so much for your reply. I've followed 1 so far. But would you please explain how I can install it via Conda? I am going to delete everything and try conda installation. If I am not mistaken, I didn't see instruction of installation via conda on github. 

Thanks

**Reply by MilesCranmer** [reply]
Create a new environment, activate it, and then `conda install -c conda-forge pysr`.

I'll add it back to the README soon. There was a problem with libstdc++ so I wanted to discourage new conda installs until that was fixed (which it seems to be)

**Reply by AKHCE** [reply]
Thank you.



It seems conda installation doesn't work. Note: I created myPySr in conda, I know that pysr is not there, but it still shows incompatibility error seemingly related to the presence of pysr in the environment I think.
